### DISCUSSION TITLE
Integrated XTTS pls? Whisper?

### DISCUSSION THREAD

**Discussion by Innomen** [LostRuins/koboldcpp#751] [Ideas]
I use pinokio for xtts but I don't see a way to link it to kobold lite. Could you build in xtts support? Maybe whipser also? Would be cool to have an all in one take to the computer and it talk's back type package heh. Thanks :)

**Comment by LostRuins** [comment]
XTTS can be selected from the TTS dropdown. Both XTTS and AllTalk are supported.

![image](https://github.com/LostRuins/koboldcpp/assets/39025047/9c9b399a-b778-4e57-b12f-d25d78b3bea9)

Get XTTS api server here: https://github.com/daswer123/xtts-api-server, run and connect to it

**Reply by Innomen** [reply]
Thank you very much, it's installing now. :)

**Reply by Innomen** [reply]
I end up with: (venv) E:\LLMs>python -m xtts_api_server
E:\LLMs\venv\Scripts\python.exe: No module named xtts_api_server
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

Claude says update visual studio build tool 2019 from 16.11.26 to 16.11.34

Will that break things? I don't wanna have to roll back visual studio, that sounds hellish.

See this is why I asked for it baked in heh, like at least some sort of optional module :)

**Reply by LostRuins** [reply]
Hmm alternatively, you can try alltalk : https://github.com/erew123/alltalk_tts

or use the colab version url https://colab.research.google.com/drive/1b-X3q5miwYLVMuiH_T73odMO8cbtICEY?usp=sharing

**Reply by Innomen** [reply]
Alltalk is emblematic of the problem with open source stuff. All that documentation and nowhere is there even a section like ? "How to run." It told me how to install it, I think it worked, (it did) ~but I can't tell because no idea how to launch it. https://github.com/erew123/alltalk_tts?tab=readme-ov-file~

(Use "stand alone")

Can be run as a [standalone](https://github.com/erew123/alltalk_tts/#-quick-setup-text-generation-webui--standalone-installation) or part of :
Text-generation-webui [link](https://github.com/oobabooga/text-generation-webui)
SillyTavern [link](https://github.com/SillyTavern/SillyTavern)
KoboldCPP [link](https://github.com/LostRuins/koboldcpp)

~So to make it work with kobold it just links back here, so here I am.~

Rant: I'm 44 and I've been online since I was 14. I did this professionally for more than a few years. It blows my mind how unfriendly all this stuff still is. I really hope AI fixes that.

~Kobold wants me to set the AllTalk API Server URL, but how do I get the server to launch? The project apparently only has an installation batch file, which as I said looks to have done its thing but no indication where it put anything or how to launch the server.~

~Super frustrating having to ask this.~

**Reply by LostRuins** [reply]
Yeah unfortunately such is the nature of the FOSS project ecosystem since each dev is working on their own project and may not have resources to ensure ease of use across all functions. Did you manage to get it to work for you?

**Comment by Innomen** [comment]
Yes I did, thank you for asking, it's a neat toy. But my machine is a potato so it's too slow. Alltalk ended up being an upgrade though, it has a nice bulk text handler. I have the option now, but it's still too much work hehe. We're still a ways off from AI ebook readers X)
